src/sentry/static/sentry/app/components/projectSelector.jsx

```diff
@@ -251,13 +255,29 @@ class ProjectSelector extends React.Component {
 
 class ProjectSelectorItem extends React.PureComponent {
   static propTypes = {
-    project: SentryTypes.Project,
+    project: SentryTypes.Project.isRequired,
+    organization: SentryTypes.Organization.isRequired,
     multi: PropTypes.bool,
     inputValue: PropTypes.string,
     isChecked: PropTypes.bool,
     onMultiSelect: PropTypes.func,
   };
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      bookmarkHasChanged: false,
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
This is deprecated, we should do this in `componentDidUpdate`

Reply:
I was told by the linter not to call `setState` in `componentDidUpdate`. Any suggestions?

Reply:
You can cheat the linter by moving the setState() into a separate method. The linter is trying to prevent infinite loops which can happen when a component updates state in DidUpdate.